### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Category.Bipointed

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Category/Bipointed.lean [hunk 1]
```diff
@@ -51,78 +56,77 @@ theorem coe_of {X : Type _} (to_prod : X × X) : ↥(of to_prod) = X :=
 #align Bipointed.coe_of Bipointed.coe_of
 
 alias of ← _root_.prod.Bipointed
```
Comment (eric-wieser) [inline Mathlib/CategoryTheory/Category/Bipointed.lean:58]:
```suggestion
alias of ← _root_.Prod.Bipointed
```

File: Mathlib/CategoryTheory/Category/Bipointed.lean [hunk 2]
```diff
@@ -27,9 +27,13 @@ universe u
 
 variable {α β : Type _}
 
+set_option linter.uppercaseLean3 false
+
 /-- The category of bipointed types. -/
 structure Bipointed : Type (u + 1) where
-  pt : Type u
+  /-- the underlying type -/
```
Comment (eric-wieser) [inline Mathlib/CategoryTheory/Category/Bipointed.lean:34]: You don't need to add these docstrings during porting

Comment (Parcly-Taxel) [reply Mathlib/CategoryTheory/Category/Bipointed.lean:34]: Aren't they needed to appease the linter?

File: Mathlib/CategoryTheory/Category/Bipointed.lean [hunk 3]
```diff
@@ -51,78 +56,77 @@ theorem coe_of {X : Type _} (to_prod : X × X) : ↥(of to_prod) = X :=
...
 #align Bipointed.hom Bipointed.Hom
 
 namespace Hom
 
 /-- The identity morphism of `X : Bipointed`. -/
 @[simps]
-def id (X : Bipointed) : Hom X X :=
-  ⟨id, rfl, rfl⟩
+def id (X : Bipointed) : Bipointed.Hom X X :=
+  ⟨_root_.id, rfl, rfl⟩
```
Comment (eric-wieser) [inline Mathlib/CategoryTheory/Category/Bipointed.lean:80]: Could also use
```suggestion
nonrec def id (X : Bipointed) : Bipointed.Hom X X :=
  ⟨id, rfl, rfl⟩
```

File: Mathlib/CategoryTheory/Category/Bipointed.lean (hunk 2 above)
Comment (eric-wieser) [reply Mathlib/CategoryTheory/Category/Bipointed.lean]: No, they're not

Comment (Parcly-Taxel) [reply Mathlib/CategoryTheory/Category/Bipointed.lean]: Good. All checks passed.

File: Mathlib/CategoryTheory/Category/Bipointed.lean [hunk 4]
```diff
@@ -27,9 +27,11 @@ universe u
 
 variable {α β : Type _}
 
+set_option linter.uppercaseLean3 false
+
 /-- The category of bipointed types. -/
 structure Bipointed : Type (u + 1) where
-  pt : Type u
+  X : Type u
```
Comment (eric-wieser) [inline Mathlib/CategoryTheory/Category/Bipointed.lean]: Is there a reason this was renamed from `pt` to `X`? If so, can you add an `#align` that records the mapping into the Lean3 name?

Comment (eric-wieser) [reply Mathlib/CategoryTheory/Category/Bipointed.lean]: Hmm, this looks like a mathport bug? It contains:

https://github.com/leanprover-community/mathlib3port/blob/master/Mathbin/CategoryTheory/Category/Bipointed.lean#L30-L34

but in Lean3 the field is called `X`. @digama0, do you know what's going on here?

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**